doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+* OtherIllegal - Any other instruction that raises an Illegal instruction exception that isn’t in the other categories.
+* None - No instruction in ID/EX stage.
+
+Stalls
+""""""
+Not all instructions can see all kinds of stalls.
+A stall category is sampled at the ID/EX stage only (as stalls in IF and WB don’t break down into categories).
+
+* Instr - A stall cause by a multi-cycle instruction.
+  This can be seen by instructions from categories:
+
+  * MUL
+  * DIV
+  * Branch
+  * Jump
+
+* LdHz - A load hazard, the instruction in ID/EX depends upon the result of a load that is awaiting a response in writeback.
+  This can be seen by instructions from categories:
+
+  * ALU
+  * Mul
+  * Div
+  * Branch
+  * Jump
+  * Load
+  * Store
+  * CSRAccess
+
+* Mem - Memory stall, the instruction in ID/EX is awaiting a prior memory request to complete before it can begin (to allow precise interrupts on a memory error response). This can be seen for all instruction categories
+
+Privilege Level
+"""""""""""""""
+
+Ibex can operate at either the M (machine) or U (user) privilege levels.
+Different aspects of the Ibex microarchitecture can be using different privilege levels at once.
+
+* Privilege level of IF stage instruction.
+* Privilege level of ID/EX stage instruction.
+* Privilege level of LSU operation (ID/EX privilege level modified by `mstatus.mprv` and `mstatus.mpp` settings).
+
+Hazards
+"""""""
+Ibex hazards all occur in the interaction between the ID and EX stage.
+
+* RAW Reg - Read after write hazard, instruction in ID/EX reads a register writeback is writing.
+  Handled with data forwarding and no stall.
+* RAW Load - Read after write hazard when reading from load destination, a special case of RAW Reg above.
+  Produces a stall (Category LdHz) and shouldn’t forward data.
+
+State Specific Behaviour
+""""""""""""""""""""""""
+Some instructions will behave differently depending upon the state of the processor (e.g. the privilege level the instruction executes at, CSR settings or whether the processor is in debug mode).
+
+* Instruction illegal in U Mode (cover execution in U and M mode).
+
+  * Mret
+  * Wfi
+  * Access to M-mode CSR.
+
+* Debug mode only instructions (cover execution in and out of debug mode).
+
+  * Dret
+  * Access to debug CSRs.
+
+    * ``dcsr``
+    * ``dpc``
+    * ``dscratch0``
+    * ``dscratch1``
+
+  * Access to trigger CSRs (also possible in M mode, cover execution in M mode, debug mode and U mode).
+
+    * ``tselect``
+    * ``tdata1``
+    * ``tdata2``
+    * ``tdata3``
+    * ``mcontext``
+    * ``scontext``
+
+* Loads/stores with ``mstatus.mprv`` set and unset.
+* EBreak behaviour in U/M mode with different ``dcsr.ebreakm`` / ``dcsr.ebreaku`` settings.
+
+Pipeline State
+^^^^^^^^^^^^^^
+Each pipeline stage has some associated state.
+
+* IF stage full or empty
+* WB stage full or empty
+* ID stage full or empty -  effectively covered by instruction categories (Category None meaning the stage is empty).
```

Comment:
```suggestion
* ID stage full or empty. This is effectively covered by instruction categories, since a category of None means the stage is empty.
```